LEAF_Request_Portal/js/formQuery.js

```diff
@@ -223,17 +222,84 @@ var LeafFormQuery = function() {
     	successCallback = funct;
     }
 
+    /**
+     * onProgress assigns a callback to be called on every getBulkData() iteration
+     * @param funct - funct(int Progress). Progress is the number of records that have been processed
+     * @memberOf LeafFormQuery
+     */
+    function onProgress(funct) {
+        progressCallback = funct;
+    }
+
+    /**
+     * Execute search query in chunks
+     * @param limitOffset Used in subsequent recursive calls to track current offset
+     * @returns Promise resolving to query response
+     * @memberOf LeafFormQuery
+     */
+    let results = {};
+    let batchSize = 500;
+    function getBulkData(limitOffset) {
+        if(limitOffset == undefined) {
+            limitOffset = 0;
+        }
+        if(limitOffset == 0) {
+            results = {};
+        }
+        limitOffset = parseInt(limitOffset);
+
+        query.limit = batchSize;
+        query.limitOffset = limitOffset;
+
+        let el = document.createElement('div');
+        el.innerHTML = JSON.stringify(query);
+        let queryUrl = el.innerText;
+
+        let dataType = 'json';
+        let urlParamJSONP = '';
+        if(useJSONP) {
+            dataType = 'jsonp';
+            urlParamJSONP = '&format=jsonp';
+        }
+
+        return $.ajax({
+            type: 'GET',
+            url: rootURL + 'api/form/query?q=' + queryUrl + extraParams + urlParamJSONP,
+            dataType: dataType
+        }).then(function(res, resStatus, resJqXHR) {
+            results = Object.assign(results, res);
+
+            if(Object.keys(res).length == batchSize
+                || resJqXHR.getResponseHeader('leaf-query') == 'continue') {
+                let newOffset = limitOffset + batchSize;
+                if(typeof progressCallback == 'function') {
+                    progressCallback(newOffset);
+                }
+                return getBulkData(newOffset);
+            }
+            else {
+                if(typeof successCallback == 'function') {
+                    successCallback(results, resStatus, resJqXHR);
+                }
+                return results;
+            }
+        });
+    }
+
     /**
      * Execute search query
-     * @param callback - Success callback
      * @returns $.ajax() object
      * @memberOf LeafFormQuery
      */
     function execute() {
     	if(query.getData != undefined && query.getData.length == 0) {
     		delete query.getData;
     	}
-    	
+
+        if(query.limit == undefined || isNaN(query.limit) || parseInt(query.limit) > 9999) {
```

Comment:
Quick question for Michael Shaffer -- were we trying to move away from truthy/falsy conditions? In this particular case we could replace 

query.limit == undefined || isNaN(query.limit) 

with 

!query.limit